### PR TITLE
Feature/cv entity

### PR DESCRIPTION
**Pull request by twentysixhugs**

### CODE REVIEW

File: src/pages/CvsPage/CvsPage.tsx [hunk 1]
```diff
@@ -18,11 +24,30 @@ const head = [
 const Table = createTable<ICVTable>();
 
 export const CvsPage = () => {
-  const cvs = useSelector((state: RootState) => state.cvs)!;
-  const dispatch = useDispatch();
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
```
Comment (tuzhyk) [inline src/pages/CvsPage/CvsPage.tsx:28]: Better to use "is" with boolean value 
isLoading
setIsLoading

File: src/pages/CvsPage/CvsPage.tsx [hunk 2]
```diff
@@ -36,15 +61,21 @@ export const CvsPage = () => {
...
-          redirectButtonText="CV details"
-          deleteButtonText="Delete"
-          entryType={TableEntry.CV}
-          showNewEntryButton={true}
-        />
+        {loading
+          ? "loader"
+          : error
+          ? "error"
+          : data?.cvs && (
```
Comment (tuzhyk) [inline src/pages/CvsPage/CvsPage.tsx:68]: Do not use nest ternary, try to create few simple constructions

File: src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import { GET_CV_INFO, UPDATE_CV } from "@graphql/Cv/Cv.queries";
+import { useEffect, useLayoutEffect, useState } from "react";
+import { SubmitHandler } from "react-hook-form";
+import { useLocation, useNavigate, useParams } from "react-router";
+import { CvInfo } from "../CvInfo/CvInfo";
+
+export const CvInfoUpdatePage = () => {
+  const { cvId } = useParams();
+
+  const [loading, setLoading] = useState(true);
```
Comment (tuzhyk) [inline src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx:17]: The same

File: src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    // TODO: Fetch projects. Show projects select component.
+    // Not a table.
+  };
+
+  return loading ? (
+    <>loader</>
+  ) : error ? (
+    <>{error}</>
+  ) : (
+    cvInput && (
```
Comment (tuzhyk) [inline src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx:102]: the same

File: src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      <CvInfo
+        cv={cvInput}
+        onSubmit={handleSubmit}
+        onCancel={() => {
+          navigate(pathname.split("/").includes("cvs") ? "/cvs" : "/employees");
+        }}
```
Comment (tuzhyk) [inline src/pages/CvsPage/components/CvInfoUpdatePage/CvInfoUpdatePage.tsx:108]: Do not use inline functions

File: src/pages/EmployeesPage/pages/EmployeeCv/EmployeeCv.tsx [hunk 6]
```diff
@@ -1,42 +1,129 @@
...
+  const [error, setError] = useState("");
+
+  const [searchParams] = useSearchParams();
+
+  const { data: userData } = useQuery<UserCvsData>(GET_USER_CVS, {
+    variables: { id: employeeId },
+    onCompleted: (data) => {
+      const firstCv = data.user.cvs[0];
+
+      if ((firstCv && firstCv.id === cvId) || (firstCv && !cvId)) {
```
Comment (tuzhyk) [inline src/pages/EmployeesPage/pages/EmployeeCv/EmployeeCv.tsx:32]: if (firstCv && (firstCv.id === cvId || !cvId)) {

File: src/pages/CvsPage/CvsPage.tsx (hunk 2 above)
Comment (twentysixhugs) [reply src/pages/CvsPage/CvsPage.tsx:68]: At first, I did it exactly the way you said. After a review, Rostik said I had to rewrite it with nested ternary. Perhaps, it is a matter of preference?

Comment (tuzhyk) [reply src/pages/CvsPage/CvsPage.tsx]: Yea, it's not critical
I prefere simple constructions, Rostik prefere nested ternary - you can do it as you wish)

**Review by tuzhyk — APPROVED** [review]

**Pull request merged**